### PR TITLE
Feat/how docs work doc

### PR DESCRIPTION
**Pull request by saltcod**
— Adds a how-docs-work doc
— Cleans up a couple of unneeded files
— removes the `libs: []` property from the client libraries common file since we're not using it

### CODE REVIEW

File: apps/docs/DEVELOPERS.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+1. id — used to identify this function
+2. title - the human-readable title
+3. slug — the url slug
+4. product - the supabase tool or product that "owns" this function. Since `updateUser()` is an auth function, it's product is `auth`
+5. type — `updateUser()` is a function and marked as such, but we can also have sections of markdown interspersed with these function definitions.
+
+#### Function Parameters
+
+The `updateUser()` function takes one parameter: `attributes`. The details for this parameter live in the GoTrue client library, referenced via a `$ref` property in the `supabase-js` [spec file](https://github.com/supabase/supabase/blob/master/spec/supabase_js_v2.yml#L357). Here, the `$ref` property is pointing to the [actual function definition](https://github.com/supabase/gotrue-js/blob/master/src/GoTrueClient.ts#L590) in the `gotrue-js` library. The accepted values for the `attributes` parameter come from the [type definition](https://github.com/supabase/gotrue-js/blob/16d3deb822097e8640a3a15b94a5690b3beaf11b/src/lib/types.ts#L233).
```
Comment (dannykng) [inline apps/docs/DEVELOPERS.md:67]:
```suggestion
The `updateUser()` function takes one parameter: `attributes`. The details for this parameter live in the GoTrue client library, referenced via a `$ref` property in the `supabase-js` [spec file](https://github.com/supabase/supabase/blob/cb04d85262db6a371539dda7df9b00ba5a901e87/spec/supabase_js_v2.yml#L357). Here, the `$ref` property is pointing to the [actual function definition](https://github.com/supabase/gotrue-js/blob/2d60e79073b96ae8c97a6ce18e2601ed1e2a2712/src/GoTrueClient.ts#L590) in the `gotrue-js` library. The accepted values for the `attributes` parameter come from the [type definition](https://github.com/supabase/gotrue-js/blob/16d3deb822097e8640a3a15b94a5690b3beaf11b/src/lib/types.ts#L233).
```

File: apps/docs/DEVELOPERS.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+These reference docs are rendered by Next.JS via a dynamic route using a [`[...slug.tsx]`](https://github.com/supabase/supabase/blob/master/apps/docs/pages/reference/javascript/%5B...slug%5D.tsx). Here, we use the library [spec file](https://github.com/supabase/supabase/blob/master/apps/docs/pages/reference/javascript/%5B...slug%5D.tsx#L4) and the [common file](https://github.com/supabase/supabase/blob/master/apps/docs/pages/reference/javascript/%5B...slug%5D.tsx#L1) to output the info you see on the page.
+
+### Other reference docs
+
+The reference docs for the [Supabase Management API](https://supabase.com/docs/reference/api) and the [Supabase CLI](https://supabase.com/docs/reference/cli) are a little more straightforward than the client libraries. Both files also have a [common file](https://github.com/supabase/supabase/blob/master/spec/common-cli-sections.json) which handles things like `title`, `id` and `slug`. Both also have a spec file detailing things like parameters, descriptions and responses ([Management API](https://github.com/supabase/supabase/blob/master/spec/api_v0_openapi.json) / [CLI](https://github.com/supabase/supabase/blob/master/spec/cli_v1_commands.yaml))
```
Comment (dannykng) [inline apps/docs/DEVELOPERS.md:81]:
```suggestion
The reference docs for the [Supabase Management API](https://supabase.com/docs/reference/api) and the [Supabase CLI](https://supabase.com/docs/reference/cli) are a little more straightforward than the client libraries. Both files also have a [common file](https://github.com/supabase/supabase/blob/master/spec/common-cli-sections.json) which handles things like `title`, `id` and `slug`. Both also have a spec file detailing things like parameters, descriptions, and responses ([Management API](https://github.com/supabase/supabase/blob/master/spec/api_v0_openapi.json) / [CLI](https://github.com/supabase/supabase/blob/master/spec/cli_v1_commands.yaml))
```

File: apps/docs/DEVELOPERS.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+#### Function Examples
+
+The `updateUser()` function has three examples listed with it. The examples are stored along with the `$ref` property in the [supabase_js_v2 spec file](https://github.com/supabase/supabase/blob/master/spec/supabase_js_v2.yml).
+
+#### Rendering in Next.JS
+
+These reference docs are rendered by Next.JS via a dynamic route using a [`[...slug.tsx]`](https://github.com/supabase/supabase/blob/master/apps/docs/pages/reference/javascript/%5B...slug%5D.tsx). Here, we use the library [spec file](https://github.com/supabase/supabase/blob/master/apps/docs/pages/reference/javascript/%5B...slug%5D.tsx#L4) and the [common file](https://github.com/supabase/supabase/blob/master/apps/docs/pages/reference/javascript/%5B...slug%5D.tsx#L1) to output the info you see on the page.
```
Comment (dannykng) [inline apps/docs/DEVELOPERS.md:77]:
```suggestion
These reference docs are rendered by Next.JS via a dynamic route using a [`[...slug.tsx]`](https://github.com/supabase/supabase/blob/master/apps/docs/pages/reference/javascript/%5B...slug%5D.tsx). Here, we use the library [spec file](https://github.com/supabase/supabase/blob/bd0514553c627db8f1e8d0b3ae440ccb6759d228/apps/docs/pages/reference/javascript/%5B...slug%5D.tsx#L4) and the [common file](https://github.com/supabase/supabase/blob/bd0514553c627db8f1e8d0b3ae440ccb6759d228/apps/docs/pages/reference/javascript/%5B...slug%5D.tsx#L1) to output the info you see on the page.
```

File: apps/docs/DEVELOPERS.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
+# Developing Supabase Docs
+
+## Getting started
+
+Thanks for your interest in [Supabase docs](https://supabase.com/docs) and for wanting to contribute! Before you begin, read the
+[code of conduct](https://github.com/supabase/.github/blob/main/CODE_OF_CONDUCT.md) and check out the
+[existing issues](https://github.com/supabase/supabase/issues).
+This document describes how to set up your development environment to contribute to [Supabase docs](https://supabase.com/docs).
+
+For a complete run-down on how all of our tools work together, see the main DEVELOPERS.md. That readme describes how to get setup locally in lots of detail, including details minimum requirements, our Turborepo setup, installing packages, sharing components across projects and more. This readme deals specifically with the docs site.
```
Comment (dannykng) [inline apps/docs/DEVELOPERS.md:10]:
```suggestion
For a complete run-down on how all of our tools work together, see the main DEVELOPERS.md. That readme describes how to get set up locally in lots of detail, including minimum requirements, our Turborepo setup, installing packages, sharing components across projects, and more. This readme deals specifically with the docs site.
```

**Review by dannykng — APPROVED** [review]

File: apps/docs/DEVELOPERS.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+#### Things to know
+
+Here's a simple [example](https://supabase.com/docs/guides/functions) `.mdx` Guide, and here is [the source on Github](https://raw.githubusercontent.com/supabase/supabase/master/apps/docs/pages/guides/functions.mdx).
+
+Some things to note:
+
+1. The files need to import a Layout at the top
+2. The files need to export a `Page` at the bottom with the `<Layout>` component
+3. The files frontmatter is stored in `const meta = {}`. You should always include `id`, `title` and `description`.
```
[the remark this answers is not in the document]
Comment (saltcod) [reply apps/docs/DEVELOPERS.md]: Yeah true. I don't think we're using these any more, but I should double check. We should clean these up....prob not in this PR though.

**Review by dannykng — APPROVED** [review]

**Pull request merged**